src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Diagnostics/RazorTranslateDiagnosticsService.cs

```diff
@@ -241,6 +241,7 @@ private static bool ShouldFilterHtmlDiagnosticBasedOnErrorCode(LspDiagnostic dia
         {
             CSSErrorCodes.UnrecognizedBlockType => IsEscapedAtSign(diagnostic, sourceText),
             CSSErrorCodes.MissingOpeningBrace => IsCSharpInStyleBlock(diagnostic, sourceText, syntaxTree),
+            CSSErrorCodes.MissingClassNameAfterDot => IsCSharpInStyleBlock(diagnostic, sourceText, syntaxTree),
```

Comment:
>IsCSharpInStyleBlock [](http://example.com/codeflow?start=54&length=20)

Probably a good enough check, this could definitely filter out legitimate errors though

Reply:
What legitimate scenarios were you thinking of? I can't think of anything, other someone than going out of their way to specify whitespace in C#, like `.@("   ")` or something,

Reply:
I'm probably confused. What does it do with:

<style>
    .{
        background-color: @(myColor);
    }
</style>

@code
{
    private string myColor = "red";
}

Reply:
Thank you, you've earned your wage for the year. I was going to say "The `.{` should be fine because we won't filter it out since there isn't C# there, but it turns out the `IsCSharpInStyleBlock` method is not "Is there C# at this position in the style block", but rather "is there any C# in this style block". Not sure which it's intended to be, but definitely I was expecting the former when I used it for this quick fix.

Also found a bug in cohosting that stops us getting any Html diagnostics while checking this, so double brownie points to you.

Reply:
Okay, fixed that, expanded the tests, and also added coverage for CSS025, "missing property value" which came up in your `background-color: @(myColor);` example.

Reply:
Looks great, thanks!